### PR TITLE
move import genbank/csv into middleware as a promise returning method

### PR DESCRIPTION
**Pull request by duncanmeech**
https://egf-atgc.atlassian.net/browse/EGFAD-437

Moving genbank/csv import into middleware

### CODE REVIEW

File: src/middleware/api.js [hunk 1]
```diff
@@ -230,6 +230,36 @@ export const infoQuery = (type, detail) => {
...
+ * project ID is returned and should be reloaded if the current project or opened if a new project.
+ * Promise resolves with projectId on success and rejects with statusText of xhr
+ */
+export const importGenbankOrCSV = (file, projectId) => {
+  invariant(file && file.name, 'expected a file object of the type that can be added to FormData');
+  const formData = new FormData();
+  formData.append('data', file, file.name);
+  const isCSV = file.name.toLowerCase().endsWith('.csv');
+  const uri = `/import/${isCSV ? 'csv' : 'genbank'}${projectId ? '/' + projectId : ''}`;
+  return new Promise((resolve, reject) => {
```
Comment (maxbates) [inline src/middleware/api.js]: is there any reaosn this isnt a fetch?

**Comment by maxbates** [comment]
why not fetch? ok otherwise

**Comment by maxbates** [comment]
const form = new FormData(__whatever__);
fetch(url, {
  method: "POST",
  body: form
})

**Comment by duncanmeech** [comment]
I saw that but there looked to issues and would involve possible tweaking some header. I'll revisit later time permitting

**Pull request merged**